### PR TITLE
mutli source (parallel) closeness centrality

### PR DESCRIPTION
**Pull request by mknblch**
ClosenessCentrality implementation using MultiSource-BFS. 

The branch is based on the branch _asp-on-msbfs_

### CODE REVIEW

**Comment by mknblch** [comment]
have to adapt some tests first

File: algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        farness = new AtomicIntegerArray(nodeCount);
+    }
+
+    public MSClosenessCentrality compute() {
+
+        final BfsConsumer consumer = (nodeId, depth, sourceNodeIds) -> {
+            // TODO size of sourceNodeIds
+            while (sourceNodeIds.hasNext()) {
+                sourceNodeIds.next();
+                farness.addAndGet(nodeId, depth);
```
Comment (knutwalker) [inline algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java]: so, it's basically `farness.addAndGet(nodeId, depth * len(sourceNodeIds))`?

Comment (knutwalker) [inline algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java]: if so, we can add or modify the third parameter to return the size w/o having to iterate.

File: algo/src/main/java/org/neo4j/graphalgo/impl/MSBFSAllShortestPaths.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.Iterator;
+import java.util.Spliterators;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * AllShortestPaths:
```
Comment (knutwalker) [inline algo/src/main/java/org/neo4j/graphalgo/impl/MSBFSAllShortestPaths.java]: JavaDoc and Comments are still copy&pasted from the PD-ASP and need to be updated.

File: tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathDeltaSteppingProcTest.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ public static void setup() throws KernelException {
     @AfterClass
     public static void shutdownGraph() throws Exception {
         api.shutdown();
-//        Pools.DEFAULT.shutdownNow();
+//        Pools.DEFAULT.shutdownNow();y
```
Comment (knutwalker) [inline tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathDeltaSteppingProcTest.java]: most important change, probably 😉

File: algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java (hunk 1 above)
Comment (mknblch) [inline algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java]: yes but can we have both :) ?

Comment (knutwalker) [inline algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java]: we can introduce a new interface/class to return there, should be no problem

File: tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathDeltaSteppingProcTest.java (hunk 3 above)
Comment (mknblch) [inline tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathDeltaSteppingProcTest.java]: oops

**Pull request closed without merging**